04.ls/ls.rb

```diff
@@ -1,33 +1,166 @@
 #!/usr/bin/env ruby
 # frozen_string_literal: true
 
+require 'optparse'
+require 'etc'
+require 'date'
+
 COLUMNS = 3
 BLANK = 2
 
+FILE_TYPE = {
+  '01' => 'p',
+  '02' => 'c',
+  '06' => 'b',
+  '10' => '-',
+  '12' => 'l',
+  '14' => 's',
+  '40' => 'd'
+}.freeze
+
+PERMISSION_MODE = {
+  '000' => '---',
+  '001' => '--x',
+  '010' => '-w-',
+  '011' => '-wx',
+  '100' => 'r--',
+  '101' => 'r-x',
+  '110' => 'rw-',
+  '111' => 'rwx'
+}.freeze
+
 def main
   default_files = Dir.glob('*')
-  files = ARGV.include?('-r') ? default_files.reverse : default_files
-  output(files)
+  if ARGV.include?('-l')
+    LongOption.new.output(default_files)
+  else
+    Default.new.output(default_files)
+  end
 end
 
-def build_file_grid(files)
-  remainder = files.size % COLUMNS
-  padding_count = (COLUMNS - remainder) % COLUMNS
-  padded_files = files + Array.new(padding_count, '')
-  row_count = padded_files.size / COLUMNS
-  file_grid = padded_files.each_slice(row_count).to_a.transpose
-  column_width = files.map(&:length).max
+class LongOption
+  def output(default_files)
+    files = default_files
+    puts "total #{FileInformation.new.total(files)}"
```

Comment:
クラスにする必要性があまりないので、ただのprivateメソッドにしましょう


```suggestion
    puts "total #{total(files)}"
```

Reply:
privateメソッドにしました！
ご確認よろしくお願い致します！